Add examples of using TreeView with react-router and NextJS router. (#2588)
primer/react: docs/content/TreeView.mdx

description: A hierarchical list of items where nested items can be expanded and collapsed.
---

```js

> import {TreeView} from '@primer/react'

Author: I was getting a linting error when I referenced TreeView this way. Should it get moved from `src` to `src/drafts`?

> import {TreeView} from '@primer/react'

Reviewer: Hmm, what was the lint error?

> import {TreeView} from '@primer/react'

Author: `error  TreeView not found in '@primer/react/drafts'  import/named`

> import {TreeView} from '@primer/react'

Reviewer: Hmm, I just ran pushed those changes and it doesn't look like there are any lint errors. I think we're good to go 👍 

I wonder if your local repository is set up differently. Have you run `npm run setup` locally?

</Box>
```

### With React Router

```jsx
import {Link, useMatch, useResolvedPath, navigate} from 'react-router-dom'

> import {TreeView} from '@primer/react'

Reviewer: ```suggestion
import {TreeView} from '@primer/react/drafts'
```

function LinkItem({id, to, children}) {
  const resolved = useResolvedPath(to)
  const isCurrent = useMatch({path: resolved.pathname, end: true})
  return (
    <TreeList.Item
      id={id}
      aria-current={isCurrent ? 'page' : undefined}
      current={isCurrent}
      onSelect={() => navigate(to)}
    >
      {children}
    </TreeList.Item>
  )
}

function App() {
  return (

>     <Box sx={{display: 'grid', gap: 2, maxWidth: 400}}>

Reviewer: This wrapping Box is probably not necessary for this example. The same applies to the Next.js example.
```suggestion
```

      <nav aria-label="Files">

>         <TreeList aria-label="Files">
>           <LinkItem id="src/Avatar.tsx" to="/src/Avatar.tsx">
>             Avatar.tsx
>           </LinkItem>
>           <LinkItem id="src/Button.tsx" to="/src/Button.tsx">
>             Button.tsx
>           </LinkItem>
>         </TreeList>

Reviewer: ```suggestion
        <TreeView aria-label="Files">
          <LinkItem id="src/Avatar.tsx" to="/src/Avatar.tsx">
            Avatar.tsx
          </LinkItem>
          <LinkItem id="src/Button.tsx" to="/src/Button.tsx">
            Button.tsx
          </LinkItem>
        </TreeView>
```

      </nav>
    </Box>
  )
}
```

### With Next.js

```jsx
import {useRouter} from 'next/router'
import Link from 'next/link'

> import {TreeView} from '@primer/react'

Reviewer: ```suggestion
import {TreeView} from '@primer/react/drafts'
```

  const router = useRouter()
  const isCurrent = typeof href === 'string' ? router.asPath === href : router.pathname === href.pathname
  return (
    <Link href={href} passHref>
      <TreeView.Item
        id={id}
        aria-current={isCurrent ? 'page' : false}
        current={isCurrent}
      >
        {children}
      </TreeView.Item>
    </Link>
  )
}

function App() {
  return (

>     <Box sx={{display: 'grid', gap: 2, maxWidth: 400}}>

Reviewer: ```suggestion
```

  return (
    <Box sx={{display: 'grid', gap: 2, maxWidth: 400}}>
      <nav aria-label="Files">
        <TreeList aria-label="Files">
          <LinkItem id="src/Avatar.tsx" href="/src/Avatar.tsx">
            Avatar.tsx
          </LinkItem>

>           <LinkItem 
>             id="${component}.${filetype}" 
>             href={{ 
>               pathname: "/src/[component].[filetype]",
>               query: { component, impl }
>             }}
>           >
>             {"${component}.${filetype}"}
>           </LinkItem>

Reviewer: I wonder if this will be confusing because `component` and `impl` aren't defined. What about something like this?
```suggestion
          <LinkItem 
            id="src/Button.tsx"
            // You can also pass a URL object
            // https://nextjs.org/docs/api-reference/next/link#with-url-object
            href={{ 
              pathname: "/src/[component].[filetype]",
              query: { component: 'Button', filetype: 'tsx' }
            }}
          >
            Button.tsx
          </LinkItem>
```
